src/nix/profile.cc

```diff
@@ -355,16 +359,30 @@ struct CmdProfileRemove : virtual EvalCommand, MixDefaultProfile, MixProfileElem
 
         for (size_t i = 0; i < oldManifest.elements.size(); ++i) {
             auto & element(oldManifest.elements[i]);
-            if (!matches(*store, element, i, matchers))
+            if (!matches(*store, element, i, matchers)) {
                 newManifest.elements.push_back(std::move(element));
+            } else {
+                notice("removing '%s' from profile '%s'", element.describe(),*profile);
```

Comment:
```suggestion
                notice("removing '%s' from profile", element.describe());
```

And maybe "from profile" is superfluous as well since we're in `nix profile remove`, so it's obvious where we're removing something from.